src/compiler/crystal/codegen/codegen.cr

```diff
@@ -1187,10 +1187,11 @@ module Crystal
     end
 
     def read_instance_var(node_type, type, name, value)
-      ivar = type.lookup_instance_var(name)
-      ivar_ptr = instance_var_ptr type, name, value
+      type_without_typedef = type.remove_typedef
+      ivar = type_without_typedef.lookup_instance_var(name)
+      ivar_ptr = instance_var_ptr type_without_typedef, name, value
       @last = downcast ivar_ptr, node_type, ivar.type, false
-      if type.extern?
+      if type_without_typedef.extern?
```

Comment:
This all could be a one liner `type = type.remove_typedef` but I didn't know how everyone felt about re-assigning arguments

Reply:
I think it's fine to reassign arguments, we do that all the time.